src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -462,6 +463,31 @@ public static InteractItemEvent callInteractItemEventSecondary(EntityPlayer play
         return event;
     }
 
+    public static boolean callDestructEntityEvent(Entity entity, Cause cause) {
+        if (((net.minecraft.entity.Entity) entity).isDead && (!(entity instanceof EntityLivingBase) || entity instanceof EntityArmorStand)) {
+            MessageChannel originalChannel = MessageChannel.TO_NONE;
+            Optional<User> sourceCreator = Optional.empty();
```

Comment:
@liach as the PR states, it's a WIP. don't need to comment to ingrain that every bit is a todo comment. It's useless PR noise unless warranted.